### PR TITLE
disable the delete button

### LINKED ISSUE #2491: Users shouldn't be able to delete their own accounts in user management
* **Overview of the issue**

An admin user can delete his own account in user management.

* **Motivation for or Use Case** 

After deleting his own account, the user is still signed in and can try to change his settings or password which would lead to an error because the data to update isn't there anymore.

* **Reproduce the error** 

Sign in as admin
Administration -> User management
Delete admin
Account -> Settings
Change First Name and Last Name
Save

* **Suggest a Fix** 

Disable the delete button for the signed in user in user management.

If my proposed solution is ok, I'll make a PR.

### PR DESCRIPTION
**Pull request by dimeros**
disable the delete button in case of logged in user

Fix #2491

### CODE REVIEW

**Comment by jdubois** [comment]
I wouldn't put more variables in the rootScope if this isn't necessary.
I need to check, but I'm pretty sure the current login is already available. Or maybe do a new query here to have it? I know it's doing more queries, but the rootScope isn't really a good place to hold stuff for just one screen

**Comment by deepu105** [comment]
Yes this is not the correct way

Just call the principal service on user management modules angular controller for this and set it on that local scope. The formatting also seems off, plz correct these

**Comment by dimeros** [comment]
Hmm, I thought that if I call it again though Principal service, this will be overkill. 
Also @jdubois that's exactly the point: the username of the application should be known somehow centrally. What about to archieve something like this in the future: 
image

So if you still don't agree I will save it to local scope and will redesign it later.

**Comment by deepu105** [comment]
@dimeros look at the principal service, it fetches from server only if the
identity is not available in the current angular service scope, so it
should be ok to call Principal service for this

File: app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 angular.module('<%=angularAppName%>')
-    .controller('UserManagementController', function ($scope, User, ParseLinks<% if (enableTranslation) { %>, Language<% } %>) {
+    .controller('UserManagementController', function ($scope, Principal, User, ParseLinks<% if (enableTranslation) { %>, Language<% } %>) {
         $scope.users = [];
         $scope.authorities = ["ROLE_USER", "ROLE_ADMIN"];<% if (enableTranslation) { %>
         Language.getAll().then(function (languages) {
             $scope.languages = languages;
         });<% } %>
-
+		Principal.identity().then(function(account) {
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js]: Can you check the indent here, I think its a single space short

File: app/templates/src/main/webapp/scripts/app/admin/user-management/user-management.html [hunk 2]
```diff
@@ -59,7 +59,8 @@ <h2 translate="user-management.home.title">Users</h2>
                     </button>
                     <button type="submit"
                             ui-sref="user-management.delete({login:user.login})"
-                            class="btn btn-danger btn-sm">
+                            class="btn btn-danger btn-sm"
+							ng-disabled="currentAccount.login==user.login">
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/user-management.html]: Plz check indent here also

**Comment by deepu105** [comment]
+1 as soon as you check the line comments above, its minor formatting

**Comment by deepu105** [comment]
Im gonna merge this and correct the indent

**Pull request merged**